serviceorchestration-simple/src/main/java/eu/arrowhead/serviceorchestration/jpa/service/OrchestrationJobDbService.java

```diff
@@ -0,0 +1,257 @@
+        Assert.isTrue(!Utilities.containsNull(statuses), "status list contains null element");
+
+        try {
+            return jobRepo.findAllByStatusIn(statuses);
+        } catch (final Exception ex) {
+            logger.error(ex.getMessage());
+            logger.debug(ex);
+            throw new InternalServerError("Database operation error");
+        }
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    @Transactional(rollbackFor = ArrowheadException.class)
+    public OrchestrationJob setStatus(final UUID jobId, final OrchestrationJobStatus status, final String message) {
+        logger.debug("setStatus started...");
+        Assert.notNull(jobId, "jobId is null");
+        Assert.notNull(status, "status is null");
+        Assert.isTrue(status != OrchestrationJobStatus.PENDING, "status can't be changed to PENDING");
+
+        try {
+            final Optional<OrchestrationJob> optional = jobRepo.findById(jobId);
+            Assert.isTrue(optional.isPresent(), "job does not exists: " + jobId);
+            final OrchestrationJob job = optional.get();
+
+            job.setStatus(status);
+            if (!Utilities.isEmpty(message)) {
+                job.setMessage(message);
+            }
+
+            switch (status) {
+                case IN_PROGRESS:
+                    job.setStartedAt(Utilities.utcNow());
+                    job.setFinishedAt(null);
+                    break;
+
+                case DONE:
+                case ERROR:
+                    job.setFinishedAt(Utilities.utcNow());
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Unhandled orchestration job status: " + status);
+            }
+
+            return jobRepo.saveAndFlush(job);
+
+        } catch (final IllegalArgumentException ex) {
+            throw ex;
+
+        } catch (final Exception ex) {
+            logger.error(ex.getMessage());
+            logger.debug(ex);
+            throw new InternalServerError("Database operation error");
+        }
+    }
+
+    //-------------------------------------------------------------------------------------------------
+    @SuppressWarnings("checkstyle:ParameterNumber")
+    public Page<OrchestrationJob> query(
+            final List<UUID> ids,
+            final List<OrchestrationJobStatus> statuses,
+            final OrchestrationType type,
+            final List<String> requesterSystems,
+            final List<String> targetSystems,
+            final List<String> serviceDefinitions,
+            final List<UUID> subscriptionIds,
+            final PageRequest pagination) {
+        logger.debug("query started...");
+        Assert.notNull(pagination, "pagination is null");
+
+        BaseFilter baseFilter = BaseFilter.NONE;
+
+        if (!Utilities.isEmpty(ids)) {
+            baseFilter = BaseFilter.ID;
+        }
+        if (!Utilities.isEmpty(statuses)) {
```

Comment:
else if (4x), because this one is always set the baseFilter, if every filter is filled (and the result will be SERVICE).